### ISSUE TITLE
Failed to get session from cache: `IllegalStateException: Daemon thread is already dead`

### ISSUE THREAD

**Issue by kobevaliant** [ktorio/ktor#1922]
Hi
Recently I encountered  this exception several times, I thought it may be a bug about session.

![1](https://user-images.githubusercontent.com/5119651/83732329-778c6080-a67e-11ea-8ad7-e62e50f8b62d.png)

**Comment by e5l** [comment]
Hi @kobevaliant, thanks for the report.
Could you share the snippet to help reproduce the problem?

**Comment by kobe2000** [comment]
Hi
I used session like following, which is very simple
```
// session object
data class FfpSession(val userId: String)

//install session
install(Sessions) {
        cookie<FfpSession>(
            "SESSION_ID",
            directorySessionStorage(File(".sessions"), cached = true)
        ) {
            cookie.path = "/"
        }
}

//use
val session = sessions.get<FfpSession>()
```

**Comment by oleg-larshin** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/KTOR-353">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.

**Comment by e5l** [comment]
Fixed since 1.5.0
